source/postprocess/geoid.cc

```diff
@@ -524,116 +609,114 @@ namespace aspect
                                                         << "degree order cosine_coefficient sine_coefficient"
                                                         << std::endl;
 
-              // write out the data on processor 0
+              // Write out the data on processor 0.
               density_anomaly_contribution_SH_coes_file << output_density_anomaly_contribution_SH_coes.str();
             }
         }
 
-      if (include_dynamic_topo_contribution == true)
+      // The user can get the spherical harmonic coefficients of the surface topography contribution if needed
+      if (also_output_surface_topo_contribution_SH_coes == true)
         {
-          // The user can get the spherical harmonic coefficients of the surface dynamic topography contribution if needed
-          if (also_output_surface_dynamic_topo_contribution_SH_coes == true)
-            {
-              // have a stream into which we write the SH coefficients data from surface dynamic topography contribution.
-              // The text stream is then later sent to processor 0
-              std::ostringstream output_surface_dynamic_topo_contribution_SH_coes;
+          // Have a stream into which we write the SH coefficients data from surface topography contribution.
+          // The text stream is then later sent to processor 0.
+          std::ostringstream output_surface_topo_contribution_SH_coes;
 
-              // Prepare the output SH coefficients data from surface dynamic topography contribution.
-              unsigned int SH_coes_ind = 0;
-              for (unsigned int ideg =  min_degree; ideg < max_degree+1; ideg++)
+          // Prepare the output SH coefficients data from surface topography contribution.
+          unsigned int SH_coes_ind = 0;
+          for (unsigned int ideg =  min_degree; ideg < max_degree+1; ideg++)
```

Comment:
please change `ideg++` to `++ideg`. The details of this change are not relevant here (but if you want you can look them up as pre-increment and post-increment operators), but since you changed the line, please fix this as well here and in the line below.